storm.yaml

```diff
@@ -4,27 +4,36 @@
 mesos.master.url: "zk://localhost:2181/mesos"
 storm.zookeeper.servers:
     - "localhost"
-nimbus.host: "localhost"
 # -----------------------------------------------------------
 
 
-### If this is not false then you need to add about 128m of mem usage to executor to account for the logviewer
-#supervisor.autostart.logviewer: false
-topology.mesos.executor.mem.mb: 1128  ## default is 1000
+# Worker resources
+topology.mesos.worker.cpu: 1.0
+### Worker heap with 20% overhead
+topology.mesos.executor.mem.mb: 1250
+worker.childopts: "-Xmx1000m"
 
-# The default behavior is to launch the logvierwer unless autostart is false
-logviewer.port: 8000
+# Supervisor resources
+topology.mesos.executor.cpu: 0.1
+topology.mesos.executor.mem.mb: 500 # Supervisor memory, with 20% overhead
+supervisor.childopts: "-Xmx256m"
+
+# The default behavior is to launch the logviewer unless autostart is false.
+# If you enable the logviewer, you'll need to add memory overhead to the
+# executor for the logviewer.
 logviewer.childopts: "-Xmx128m"
 logviewer.cleanup.age.mins: 10080
 logviewer.appender.name: "A1"
-
-
-# You should not need to change anything below this line
-#--------------------------------------------------------
+supervisor.autostart.logviewer: true
```

Comment:
Ah... I haven't read everything you've changed here.  But I have now skimmed through the logviewer parts -- you seem to be grabbing another port for per-executor/supervisor logviewers and launching them appropriately.  But as you just guessed, there is no interaction between the nimbus nor storm-UI and the logviewer about what port the logviewer has on a particular host for a particular topology.   From grepping the storm master branch, we can see that it is just 1 global config knob for the logviewers' port:

```
storm-core/src/clj/backtype/storm/daemon/logviewer.clj:          https-port (int (or (conf LOGVIEWER-HTTPS-PORT) 0))
storm-core/src/clj/backtype/storm/daemon/logviewer.clj:      (storm-run-jetty {:port (int (conf LOGVIEWER-PORT))
storm-core/src/clj/backtype/storm/ui/core.clj:  (if (and secure? (*STORM-CONF* LOGVIEWER-HTTPS-PORT))
storm-core/src/clj/backtype/storm/ui/core.clj:      (*STORM-CONF* LOGVIEWER-HTTPS-PORT)
storm-core/src/clj/backtype/storm/ui/core.clj:      (*STORM-CONF* LOGVIEWER-PORT)
storm-core/src/clj/backtype/storm/ui/core.clj:  (logviewer-link host (event-logs-filename topology-id port) secure?))
storm-core/src/clj/backtype/storm/ui/core.clj:  (url-format "http://%s:%s/daemonlog?file=nimbus.log" host (*STORM-CONF* LOGVIEWER-PORT) port))
storm-core/src/clj/backtype/storm/ui/core.clj:              (*STORM-CONF* LOGVIEWER-PORT)
storm-core/src/jvm/backtype/storm/Config.java:    public static final String LOGVIEWER_PORT = "logviewer.port";
storm-core/src/jvm/backtype/storm/Config.java:     * Storm Logviewer HTTPS port
storm-core/src/jvm/backtype/storm/Config.java:    public static final String LOGVIEWER_HTTPS_PORT = "logviewer.https.port";
```